starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -2619,6 +2619,19 @@ sub quat2radecroll {
     return ($ra, $dec, $roll);
 }
 
+###################################################################################
+sub check_guide_count{
+###################################################################################
+    my $self = shift;
+    my $guide_count = $self->count_guide_stars();
+
+    if ($guide_count < 4.0){
+        push @{$self->{warn}}, "Guide count of $guide_count < 4.0.";
+    }
+
+    # Also save the guide count in the figure_of_merit
+    $self->{figure_of_merit}->{guide_count} = $guide_count;
```

Comment:
I'm not thrilled by coding with "side-effects" like this code that also saves the guide count into the data structure.  But it does seem pretty consistent with other pieces of starcheck.

Reply:
I'm confused about this new checking method since I thought the scope here was basically just formatting and updated output.

Put in another way, isn't there already code that is checking for guide count < 4.0?

Reply:
So for that, just a question about if we want to call the guide count calculation method twice.

Reply:
I'm not worried about doing the guide count calculation twice since it is not very expensive, so reducing any coupling / complexity is a good thing.

Having now looked at the code I realize what is obvious to you (and makes me remember now) that there was no check at the obsid-level for too few guide stars. This only happened in the summary code. That was really my question.

But now looking at this, doesn't the 4.0 limit need to be 6.0 for ERs?  Ala
```
        my $min_num_gui = ($obs{$obsid}->{obsid} >= 38000 ) ? 6.0 : 4.0;
```

Reply:
I think we still want to save the value in the figure of merit hash to make it available for printing.  So, not sure if it is worth it to make a new call to the counting code or not at the point of the summary check.